src/app/pages/splash/services/splash.service.ts

```diff
@@ -12,15 +13,20 @@ export class SplashService {
     private config: ConfigService,
     private token: TokenService,
     private schedule: ScheduleService,
-    private usage: UsageService
+    private usage: UsageService,
+    private notificationService: NotificationService
   ) {}
 
   evalEnrolment() {
     return this.token.isValid().catch(() => false)
   }
 
   loadConfig() {
-    return this.token.refresh().then(() => this.config.fetchConfigState())
+    return this.notificationService
+      .init()
+      .then(() => this.notificationService.permissionCheck())
+      .then(() => this.token.refresh())
```

Comment:
shouldn't the token be refreshed first since it will try to register project and subject with appserver on init?